### ISSUE TITLE
A loaded model seems not inference properly

### ISSUE THREAD

**Issue by seungickjang** [NVIDIA/Megatron-LM#65]
Hello,

I fine-tuned a pre-trained BERT with RACE data and tried to inference. 

While fine-tuning the BERT, the accuracy rate was about 60~70% with the valid data. But during the inference, the accuracy rate on the valid data was only about 25% which is almost random.

When I saved/loaded a model, I used 'save_checkpoint' and 'load_checkpoint' functions in the megatron/checkingpoint.py, and called model.eval() for evaluation as well. Also, I referenced "tasks/zeroshot_gpt2/evaluate.py" and modified for my own purpose.

Is it a bug? Or it would be good if there's any example code or documentation regarding this issue.

Regards,
